### ISSUE TITLE
Offline Support

### ISSUE THREAD

**Issue by OR13** [w3c-ccg/webkms#5]
I've been exploring the magic of service workers, and I believe its possible to implement a webkms entirely in browser, with the http interface intercepted by service workers, such that it will work in offline environments. 

Would it be acceptable to use this repo with https://github.com/digitalbazaar/webkms-client to demonstrate this functionality?

**Comment by dlongley** [comment]
We might also want to specify a JavaScript API to allow local KMS modules to be registered and used -- and these could be used in offline mode without the need for service worker interception. I'm not quite sure how useful service worker interception would be other than as a redirection mechanism to some local KMS module anyway -- and a direct JS API would be better suited for that.

**Comment by dlongley** [comment]
Experiments welcome, of course.

**Comment by OR13** [comment]
Yes, I agree a local KMS JS API would be better, however, it would probably be more work than simply taking the existing API and building a service worker backed http api... since we can reuse the entire webkms-client, instead of needing to a whole new local client... I'm eager to get this connected to non extractable browser keys, but I don't have enough time or skill to create a brand new client, and its probably not a good idea if the interface is not stable now anyway... if we still to HTTP, it will be easier for us to align i think.
